packages/framework-core/src/booster-graphql-dispatcher.ts

```diff
@@ -49,7 +48,8 @@ export class BoosterGraphQLDispatcher {
       case 'CONNECT':
         return this.config.provider.graphQL.handleResult(null, graphQLWebsocketSubprotocolHeaders)
       case 'MESSAGE':
-        return this.config.provider.graphQL.handleResult(await this.handleMessage(envelopeOrError))
+        const result = await this.handleMessage(envelopeOrError)
+        return this.config.provider.graphQL.handleResult(result.graphQLResult, result.responseHeaders)
```

Comment:
I think this is 100% fine, but it seems that you're doing nothing with the response headers inside of the `handleMessage` and the other methods. Would it make sense to create the `responseHeaders` instance here instead?

```suggestion
        const responseHeaders = { ...this.config.defaultResponseHeaders }
        const result = await this.handleMessage(envelopeOrError, responseHeaders)
        return this.config.provider.graphQL.handleResult(result, responseHeaders)
```

Doing this also requires passing this object through all functions, so not sure if this will produce any cleaner code...